In [1]:
import pandas as pd
import pathlib

In [2]:
dataset_path = pathlib.Path.cwd()/'Datasets'
WHS_path = dataset_path/'World_Health_Statistics_archive'

## Cancer

In [3]:
cancer_data = pd.read_csv(WHS_path/'30-70cancerChdEtc.csv')

In [4]:
cancer_data

Location  Period                                          Indicator  \
0     Afghanistan    2016  Probability (%) of dying between age 30 and ex...   
1     Afghanistan    2016  Probability (%) of dying between age 30 and ex...   
2     Afghanistan    2016  Probability (%) of dying between age 30 and ex...   
3     Afghanistan    2015  Probability (%) of dying between age 30 and ex...   
4     Afghanistan    2015  Probability (%) of dying between age 30 and ex...   
...           ...     ...                                                ...   
2740     Zimbabwe    2005  Probability (%) of dying between age 30 and ex...   
2741     Zimbabwe    2005  Probability (%) of dying between age 30 and ex...   
2742     Zimbabwe    2000  Probability (%) of dying between age 30 and ex...   
2743     Zimbabwe    2000  Probability (%) of dying between age 30 and ex...   
2744     Zimbabwe    2000  Probability (%) of dying between age 30 and ex...   

            Dim1  First Tooltip  
0     Both sexes           29.8  
1           Male           31.8  
2         Female           27.7  
3     Both sexes           29.8  
4           Male           31.9  
...          ...            ...  
2740        Male           22.1  
2741      Female           22.9  
2742  Both sexes           21.6  
2743        Male           21.6  
2744      Female           21.7  

[2745 rows x 5 columns]

There is no nulls in this dataset

In [5]:
cancer_data.isnull().count()

Location         2745
Period           2745
Indicator        2745
Dim1             2745
First Tooltip    2745
dtype: int64

The years in this dataset are 2000, 2005, 2010, 2015, and 2016. It appears to be every 5 years except in regards to 2016.

In [6]:
cancer_data['Period'].unique()

array([2016, 2015, 2010, 2005, 2000])

All counts are the same for each year, so any year is a great choice.

In [7]:
cancer_data[cancer_data['Period'] == 2016].count()

Location         549
Period           549
Indicator        549
Dim1             549
First Tooltip    549
dtype: int64

In [8]:
cancer_data[cancer_data['Period'] == 2015].count()

Location         549
Period           549
Indicator        549
Dim1             549
First Tooltip    549
dtype: int64

In [9]:
cancer_data[cancer_data['Period'] == 2010].count()

Location         549
Period           549
Indicator        549
Dim1             549
First Tooltip    549
dtype: int64

In [10]:
cancer_data[cancer_data['Period'] == 2005].count()

Location         549
Period           549
Indicator        549
Dim1             549
First Tooltip    549
dtype: int64

In [11]:
cancer_data[cancer_data['Period'] == 2000].count()

Location         549
Period           549
Indicator        549
Dim1             549
First Tooltip    549
dtype: int64

All data is about dying between the age 30 and 70 from cardiovascular disease, cancer, diabetes, or a chronic respiratory disease.

In [12]:
cancer_data['Indicator'].unique()

array(['Probability (%) of dying between age 30 and exact age 70 from any of cardiovascular disease, cancer, diabetes, or chronic respiratory disease'],
      dtype=object)

In [13]:
#cancer_data['Dim1'].str.contains("Both Sexes").count()

In [14]:
#cancer_data.iloc[[0]]

In [15]:
cancer_both_sex = cancer_data[cancer_data['Dim1'] == 'Both sexes']
cancer_both_sex

Location  Period                                          Indicator  \
0     Afghanistan    2016  Probability (%) of dying between age 30 and ex...   
3     Afghanistan    2015  Probability (%) of dying between age 30 and ex...   
6     Afghanistan    2010  Probability (%) of dying between age 30 and ex...   
9     Afghanistan    2005  Probability (%) of dying between age 30 and ex...   
12    Afghanistan    2000  Probability (%) of dying between age 30 and ex...   
...           ...     ...                                                ...   
2730     Zimbabwe    2016  Probability (%) of dying between age 30 and ex...   
2733     Zimbabwe    2015  Probability (%) of dying between age 30 and ex...   
2736     Zimbabwe    2010  Probability (%) of dying between age 30 and ex...   
2739     Zimbabwe    2005  Probability (%) of dying between age 30 and ex...   
2742     Zimbabwe    2000  Probability (%) of dying between age 30 and ex...   

            Dim1  First Tooltip  
0     Both sexes           29.8  
3     Both sexes           29.8  
6     Both sexes           31.7  
9     Both sexes           34.1  
12    Both sexes           34.4  
...          ...            ...  
2730  Both sexes           19.3  
2733  Both sexes           19.4  
2736  Both sexes           21.1  
2739  Both sexes           22.5  
2742  Both sexes           21.6  

[915 rows x 5 columns]

## This dataframe will need to be exported.

In [16]:
cancer_both_sex_2016 = cancer_both_sex[cancer_both_sex['Period'] == 2016]
cancer_both_sex_2016

Location  Period  \
0                            Afghanistan    2016   
15                               Albania    2016   
30                               Algeria    2016   
45                                Angola    2016   
60                   Antigua and Barbuda    2016   
...                                  ...     ...   
2670  Venezuela (Bolivarian Republic of)    2016   
2685                            Viet Nam    2016   
2700                               Yemen    2016   
2715                              Zambia    2016   
2730                            Zimbabwe    2016   

                                              Indicator        Dim1  \
0     Probability (%) of dying between age 30 and ex...  Both sexes   
15    Probability (%) of dying between age 30 and ex...  Both sexes   
30    Probability (%) of dying between age 30 and ex...  Both sexes   
45    Probability (%) of dying between age 30 and ex...  Both sexes   
60    Probability (%) of dying between age 30 and ex...  Both sexes   
...                                                 ...         ...   
2670  Probability (%) of dying between age 30 and ex...  Both sexes   
2685  Probability (%) of dying between age 30 and ex...  Both sexes   
2700  Probability (%) of dying between age 30 and ex...  Both sexes   
2715  Probability (%) of dying between age 30 and ex...  Both sexes   
2730  Probability (%) of dying between age 30 and ex...  Both sexes   

      First Tooltip  
0              29.8  
15             17.0  
30             14.2  
45             16.5  
60             22.6  
...             ...  
2670           18.1  
2685           17.1  
2700           30.6  
2715           17.9  
2730           19.3  

[183 rows x 5 columns]

## This dataframe will need to be exported.

In [17]:
cancer_both_sex_2015 = cancer_both_sex[cancer_both_sex['Period'] == 2015]
cancer_both_sex_2015

Location  Period  \
3                            Afghanistan    2015   
18                               Albania    2015   
33                               Algeria    2015   
48                                Angola    2015   
63                   Antigua and Barbuda    2015   
...                                  ...     ...   
2673  Venezuela (Bolivarian Republic of)    2015   
2688                            Viet Nam    2015   
2703                               Yemen    2015   
2718                              Zambia    2015   
2733                            Zimbabwe    2015   

                                              Indicator        Dim1  \
3     Probability (%) of dying between age 30 and ex...  Both sexes   
18    Probability (%) of dying between age 30 and ex...  Both sexes   
33    Probability (%) of dying between age 30 and ex...  Both sexes   
48    Probability (%) of dying between age 30 and ex...  Both sexes   
63    Probability (%) of dying between age 30 and ex...  Both sexes   
...                                                 ...         ...   
2673  Probability (%) of dying between age 30 and ex...  Both sexes   
2688  Probability (%) of dying between age 30 and ex...  Both sexes   
2703  Probability (%) of dying between age 30 and ex...  Both sexes   
2718  Probability (%) of dying between age 30 and ex...  Both sexes   
2733  Probability (%) of dying between age 30 and ex...  Both sexes   

      First Tooltip  
3              29.8  
18             17.4  
33             14.4  
48             16.9  
63             22.6  
...             ...  
2673           18.2  
2688           17.3  
2703           30.7  
2718           18.0  
2733           19.4  

[183 rows x 5 columns]

## Adolescent Birth Rate

In [18]:
ad_br_data = pd.read_csv(WHS_path/'adolescentBirthRate.csv')
ad_br_data

Location  Period                                          Indicator  \
0     Afghanistan    2017  Adolescent birth rate (per 1000 women aged 15-...   
1     Afghanistan    2014  Adolescent birth rate (per 1000 women aged 15-...   
2     Afghanistan    2013  Adolescent birth rate (per 1000 women aged 15-...   
3     Afghanistan    2011  Adolescent birth rate (per 1000 women aged 15-...   
4     Afghanistan    2009  Adolescent birth rate (per 1000 women aged 15-...   
...           ...     ...                                                ...   
2184     Zimbabwe    2008  Adolescent birth rate (per 1000 women aged 15-...   
2185     Zimbabwe    2007  Adolescent birth rate (per 1000 women aged 15-...   
2186     Zimbabwe    2004  Adolescent birth rate (per 1000 women aged 15-...   
2187     Zimbabwe    2003  Adolescent birth rate (per 1000 women aged 15-...   
2188     Zimbabwe    2002  Adolescent birth rate (per 1000 women aged 15-...   

      First Tooltip  
0              62.0  
1              77.2  
2              87.0  
3             125.7  
4              80.0  
...             ...  
2184          112.0  
2185          102.6  
2186           98.7  
2187          101.0  
2188          102.6  

[2189 rows x 4 columns]

In [19]:
ad_br_data.isnull().count()

Location         2189
Period           2189
Indicator        2189
First Tooltip    2189
dtype: int64

In [20]:
ad_br_data['Period'].unique()

array([2017, 2014, 2013, 2011, 2009, 2008, 2007, 2006, 2005, 2003, 2000,
       2016, 2015, 2012, 2010, 2004, 2002, 2001, 2018])

Data counts are all different for each year. This will need to be evaluated further since 2015 and 2016 might not be the best options.

In [21]:
ad_br_data[ad_br_data['Period'] == 2018].count()

Location         22
Period           22
Indicator        22
First Tooltip    22
dtype: int64

In [22]:
ad_br_data[ad_br_data['Period'] == 2017].count()

Location         99
Period           99
Indicator        99
First Tooltip    99
dtype: int64

In [23]:
ad_br_data[ad_br_data['Period'] == 2016].count()

Location         108
Period           108
Indicator        108
First Tooltip    108
dtype: int64

In [24]:
ad_br_data[ad_br_data['Period'] == 2015].count()

Location         106
Period           106
Indicator        106
First Tooltip    106
dtype: int64

In [25]:
ad_br_data[ad_br_data['Period'] == 2014].count()

Location         115
Period           115
Indicator        115
First Tooltip    115
dtype: int64

In [26]:
ad_br_data[ad_br_data['Period'] == 2013].count()

Location         116
Period           116
Indicator        116
First Tooltip    116
dtype: int64

In [27]:
ad_br_data[ad_br_data['Period'] == 2012].count()

Location         113
Period           113
Indicator        113
First Tooltip    113
dtype: int64

In [28]:
ad_br_data[ad_br_data['Period'] == 2011].count()

Location         117
Period           117
Indicator        117
First Tooltip    117
dtype: int64

In [29]:
ad_br_data[ad_br_data['Period'] == 2010].count()

Location         120
Period           120
Indicator        120
First Tooltip    120
dtype: int64

In [30]:
ad_br_data[ad_br_data['Period'] == 2009].count()

Location         133
Period           133
Indicator        133
First Tooltip    133
dtype: int64

In [31]:
ad_br_data[ad_br_data['Period'] == 2008].count()

Location         116
Period           116
Indicator        116
First Tooltip    116
dtype: int64

In [32]:
ad_br_data[ad_br_data['Period'] == 2007].count()

Location         121
Period           121
Indicator        121
First Tooltip    121
dtype: int64

In [33]:
ad_br_data[ad_br_data['Period'] == 2006].count()

Location         120
Period           120
Indicator        120
First Tooltip    120
dtype: int64

In [34]:
ad_br_data[ad_br_data['Period'] == 2005].count()

Location         130
Period           130
Indicator        130
First Tooltip    130
dtype: int64

In [35]:
ad_br_data[ad_br_data['Period'] == 2004].count()

Location         124
Period           124
Indicator        124
First Tooltip    124
dtype: int64

In [36]:
ad_br_data[ad_br_data['Period'] == 2003].count()

Location         120
Period           120
Indicator        120
First Tooltip    120
dtype: int64

In [37]:
ad_br_data[ad_br_data['Period'] == 2002].count()

Location         134
Period           134
Indicator        134
First Tooltip    134
dtype: int64

In [38]:
ad_br_data[ad_br_data['Period'] == 2001].count()

Location         133
Period           133
Indicator        133
First Tooltip    133
dtype: int64

In [39]:
ad_br_data[ad_br_data['Period'] == 2000].count()

Location         142
Period           142
Indicator        142
First Tooltip    142
dtype: int64

In [40]:
ad_br_data['Indicator'].unique()

array(['Adolescent birth rate (per 1000 women aged 15-19 years)'],
      dtype=object)

## Check if use this year

This year only has 108 rows

In [41]:
ad_br_2016 = ad_br_data[ad_br_data['Period'] == 2016]
ad_br_2016

Location  Period  \
12                                              Albania    2016   
41                                              Andorra    2016   
66                                  Antigua and Barbuda    2016   
96                                              Armenia    2016   
114                                           Australia    2016   
...                                                 ...     ...   
2065  United Kingdom of Great Britain and Northern I...    2016   
2082                        United Republic of Tanzania    2016   
2092                           United States of America    2016   
2111                                            Uruguay    2016   
2129                                         Uzbekistan    2016   

                                              Indicator  First Tooltip  
12    Adolescent birth rate (per 1000 women aged 15-...           17.3  
41    Adolescent birth rate (per 1000 women aged 15-...            2.6  
66    Adolescent birth rate (per 1000 women aged 15-...           31.1  
96    Adolescent birth rate (per 1000 women aged 15-...           24.5  
114   Adolescent birth rate (per 1000 women aged 15-...           10.3  
...                                                 ...            ...  
2065  Adolescent birth rate (per 1000 women aged 15-...           13.7  
2082  Adolescent birth rate (per 1000 women aged 15-...          138.9  
2092  Adolescent birth rate (per 1000 women aged 15-...           20.3  
2111  Adolescent birth rate (per 1000 women aged 15-...           50.3  
2129  Adolescent birth rate (per 1000 women aged 15-...           20.9  

[108 rows x 4 columns]

## Check if use this year

This row has only 106 rows

In [42]:
ad_br_2015 = ad_br_data[ad_br_data['Period'] == 2015]
ad_br_2015

Location  Period  \
13                                              Albania    2015   
28                                              Algeria    2015   
42                                              Andorra    2015   
67                                  Antigua and Barbuda    2015   
79                                            Argentina    2015   
...                                                 ...     ...   
2000                                             Turkey    2015   
2027                                             Uganda    2015   
2066  United Kingdom of Great Britain and Northern I...    2015   
2093                           United States of America    2015   
2112                                            Uruguay    2015   

                                              Indicator  First Tooltip  
13    Adolescent birth rate (per 1000 women aged 15-...           20.2  
28    Adolescent birth rate (per 1000 women aged 15-...            9.7  
42    Adolescent birth rate (per 1000 women aged 15-...            3.1  
67    Adolescent birth rate (per 1000 women aged 15-...           37.7  
79    Adolescent birth rate (per 1000 women aged 15-...           65.1  
...                                                 ...            ...  
2000  Adolescent birth rate (per 1000 women aged 15-...           24.9  
2027  Adolescent birth rate (per 1000 women aged 15-...          131.5  
2066  Adolescent birth rate (per 1000 women aged 15-...           14.4  
2093  Adolescent birth rate (per 1000 women aged 15-...           22.3  
2112  Adolescent birth rate (per 1000 women aged 15-...           55.6  

[106 rows x 4 columns]

## Air Pollution Deaths

In [43]:
ap_data = pd.read_csv(WHS_path/'airPollutionDeathRate.csv')
ap_data

Location                                   Dim2  \
0     Afghanistan                                  Total   
1     Afghanistan                                  Total   
2     Afghanistan                                  Total   
3     Afghanistan                                  Total   
4     Afghanistan                                  Total   
...           ...                                    ...   
6583     Zimbabwe  Chronic obstructive pulmonary disease   
6584     Zimbabwe  Chronic obstructive pulmonary disease   
6585     Zimbabwe  Chronic obstructive pulmonary disease   
6586     Zimbabwe  Chronic obstructive pulmonary disease   
6587     Zimbabwe  Chronic obstructive pulmonary disease   

                                              Indicator  Period        Dim1  \
0     Ambient and household air pollution attributab...    2016  Both sexes   
1     Ambient and household air pollution attributab...    2016        Male   
2     Ambient and household air pollution attributab...    2016      Female   
3     Ambient and household air pollution attributab...    2016  Both sexes   
4     Ambient and household air pollution attributab...    2016        Male   
...                                                 ...     ...         ...   
6583  Ambient and household air pollution attributab...    2016        Male   
6584  Ambient and household air pollution attributab...    2016      Female   
6585  Ambient and household air pollution attributab...    2016  Both sexes   
6586  Ambient and household air pollution attributab...    2016        Male   
6587  Ambient and household air pollution attributab...    2016      Female   

            First Tooltip  
0     95.12 [85.12-104.3]  
1        95.58 [85.5-106]  
2     94.63 [84.55-103.9]  
3     211.1 [188.9-231.3]  
4     228.7 [204.6-253.6]  
...                   ...  
6583     5.54 [2.74-7.66]  
6584      6.19 [4.73-7.3]  
6585  16.09 [11.67-19.89]  
6586   17.72 [8.75-24.51]  
6587  14.95 [11.43-17.62]  

[6588 rows x 6 columns]

In [44]:
ap_data.isnull().count()

Location         6588
Dim2             6588
Indicator        6588
Period           6588
Dim1             6588
First Tooltip    6588
dtype: int64

In [45]:
ap_data['Period'].unique()

array([2016])

In [46]:
ap_data['Dim2'].unique()

array(['Total', 'Lower respiratory infections',
       'Trachea, bronchus, lung cancers', 'Ischaemic heart disease',
       'Stroke', 'Chronic obstructive pulmonary disease'], dtype=object)

In [47]:
ap_data['Indicator'].unique()

array(['Ambient and household air pollution attributable death rate (per 100 000 population)',
       'Ambient and household air pollution attributable death rate (per 100 000 population, age-standardized)'],
      dtype=object)

I think it would be best to make each unique value in Dim2 as a column. That way we would have a single row for each country instead of a bunch of rows per country. Maybe see if the original dataset was set up like this.

In [48]:
ap_bs_data = ap_data[ap_data['Dim1'] == 'Both sexes']
ap_bs_data

Location                                   Dim2  \
0     Afghanistan                                  Total   
3     Afghanistan                                  Total   
6     Afghanistan           Lower respiratory infections   
9     Afghanistan           Lower respiratory infections   
12    Afghanistan        Trachea, bronchus, lung cancers   
...           ...                                    ...   
6573     Zimbabwe                Ischaemic heart disease   
6576     Zimbabwe                                 Stroke   
6579     Zimbabwe                                 Stroke   
6582     Zimbabwe  Chronic obstructive pulmonary disease   
6585     Zimbabwe  Chronic obstructive pulmonary disease   

                                              Indicator  Period        Dim1  \
0     Ambient and household air pollution attributab...    2016  Both sexes   
3     Ambient and household air pollution attributab...    2016  Both sexes   
6     Ambient and household air pollution attributab...    2016  Both sexes   
9     Ambient and household air pollution attributab...    2016  Both sexes   
12    Ambient and household air pollution attributab...    2016  Both sexes   
...                                                 ...     ...         ...   
6573  Ambient and household air pollution attributab...    2016  Both sexes   
6576  Ambient and household air pollution attributab...    2016  Both sexes   
6579  Ambient and household air pollution attributab...    2016  Both sexes   
6582  Ambient and household air pollution attributab...    2016  Both sexes   
6585  Ambient and household air pollution attributab...    2016  Both sexes   

            First Tooltip  
0     95.12 [85.12-104.3]  
3     211.1 [188.9-231.3]  
6     31.26 [25.66-35.67]  
9     31.68 [26.01-36.15]  
12       1.61 [1.21-1.96]  
...                   ...  
6573   33.24 [28.49-38.1]  
6576     8.27 [7.07-9.64]  
6579   20.08 [17.15-23.4]  
6582     5.88 [4.26-7.26]  
6585  16.09 [11.67-19.89]  

[2196 rows x 6 columns]

In [49]:
ap_bs_standardized = ap_bs_data[ap_bs_data['Indicator'] == 'Ambient and household air pollution attributable death rate (per 100 000 population, age-standardized)']
ap_bs_standardized

Location                                   Dim2  \
3     Afghanistan                                  Total   
9     Afghanistan           Lower respiratory infections   
15    Afghanistan        Trachea, bronchus, lung cancers   
21    Afghanistan                Ischaemic heart disease   
27    Afghanistan                                 Stroke   
...           ...                                    ...   
6561     Zimbabwe           Lower respiratory infections   
6567     Zimbabwe        Trachea, bronchus, lung cancers   
6573     Zimbabwe                Ischaemic heart disease   
6579     Zimbabwe                                 Stroke   
6585     Zimbabwe  Chronic obstructive pulmonary disease   

                                              Indicator  Period        Dim1  \
3     Ambient and household air pollution attributab...    2016  Both sexes   
9     Ambient and household air pollution attributab...    2016  Both sexes   
15    Ambient and household air pollution attributab...    2016  Both sexes   
21    Ambient and household air pollution attributab...    2016  Both sexes   
27    Ambient and household air pollution attributab...    2016  Both sexes   
...                                                 ...     ...         ...   
6561  Ambient and household air pollution attributab...    2016  Both sexes   
6567  Ambient and household air pollution attributab...    2016  Both sexes   
6573  Ambient and household air pollution attributab...    2016  Both sexes   
6579  Ambient and household air pollution attributab...    2016  Both sexes   
6585  Ambient and household air pollution attributab...    2016  Both sexes   

            First Tooltip  
3     211.1 [188.9-231.3]  
9     31.68 [26.01-36.15]  
15       4.05 [3.03-4.92]  
21    106.9 [91.91-121.1]  
27     45.1 [38.59-51.65]  
...                   ...  
6561  61.68 [51.05-71.05]  
6567     1.89 [1.42-2.33]  
6573   33.24 [28.49-38.1]  
6579   20.08 [17.15-23.4]  
6585  16.09 [11.67-19.89]  

[1098 rows x 6 columns]

In [50]:
ap_bs_standardized['Indicator'].unique()

array(['Ambient and household air pollution attributable death rate (per 100 000 population, age-standardized)'],
      dtype=object)

## Alcohol Substance Abuse

In [51]:
alcohol_data = pd.read_csv(WHS_path/'alcoholSubstanceAbuse.csv')
alcohol_data

Location  Period                                          Indicator  \
0     Afghanistan    2018  Total (recorded+unrecorded) alcohol per capita...   
1     Afghanistan    2018  Total (recorded+unrecorded) alcohol per capita...   
2     Afghanistan    2018  Total (recorded+unrecorded) alcohol per capita...   
3     Afghanistan    2015  Total (recorded+unrecorded) alcohol per capita...   
4     Afghanistan    2015  Total (recorded+unrecorded) alcohol per capita...   
...           ...     ...                                                ...   
2782     Zimbabwe    2005  Total (recorded+unrecorded) alcohol per capita...   
2783     Zimbabwe    2005  Total (recorded+unrecorded) alcohol per capita...   
2784     Zimbabwe    2000  Total (recorded+unrecorded) alcohol per capita...   
2785     Zimbabwe    2000  Total (recorded+unrecorded) alcohol per capita...   
2786     Zimbabwe    2000  Total (recorded+unrecorded) alcohol per capita...   

            Dim1  First Tooltip  
0     Both sexes          0.210  
1           Male          0.360  
2         Female          0.055  
3     Both sexes          0.210  
4           Male          0.350  
...          ...            ...  
2782        Male          4.720  
2783      Female          0.990  
2784  Both sexes          2.460  
2785        Male          4.160  
2786      Female          0.880  

[2787 rows x 5 columns]

In [52]:
alcohol_data.isnull().count()

Location         2787
Period           2787
Indicator        2787
Dim1             2787
First Tooltip    2787
dtype: int64

In [53]:
alcohol_data['Period'].unique()

array([2018, 2015, 2010, 2005, 2000])

The years to use are 2018 and 2015.

In [54]:
alcohol_data[alcohol_data['Period'] == 2018].count()

Location         564
Period           564
Indicator        564
Dim1             564
First Tooltip    564
dtype: int64

In [55]:
alcohol_data[alcohol_data['Period'] == 2015].count()

Location         564
Period           564
Indicator        564
Dim1             564
First Tooltip    564
dtype: int64

In [56]:
alcohol_data[alcohol_data['Period'] == 2010].count()

Location         561
Period           561
Indicator        561
Dim1             561
First Tooltip    561
dtype: int64

In [57]:
alcohol_data[alcohol_data['Period'] == 2005].count()

Location         549
Period           549
Indicator        549
Dim1             549
First Tooltip    549
dtype: int64

In [58]:
alcohol_data[alcohol_data['Period'] == 2000].count()

Location         549
Period           549
Indicator        549
Dim1             549
First Tooltip    549
dtype: int64

In [59]:
alcohol_data['Indicator'].unique()

array(['Total (recorded+unrecorded) alcohol per capita (15+) consumption'],
      dtype=object)

In [60]:
alcohol_both_sex = alcohol_data[alcohol_data['Dim1'] == 'Both sexes']
alcohol_both_sex

Location  Period                                          Indicator  \
0     Afghanistan    2018  Total (recorded+unrecorded) alcohol per capita...   
3     Afghanistan    2015  Total (recorded+unrecorded) alcohol per capita...   
6     Afghanistan    2010  Total (recorded+unrecorded) alcohol per capita...   
9         Albania    2018  Total (recorded+unrecorded) alcohol per capita...   
12        Albania    2015  Total (recorded+unrecorded) alcohol per capita...   
...           ...     ...                                                ...   
2772     Zimbabwe    2018  Total (recorded+unrecorded) alcohol per capita...   
2775     Zimbabwe    2015  Total (recorded+unrecorded) alcohol per capita...   
2778     Zimbabwe    2010  Total (recorded+unrecorded) alcohol per capita...   
2781     Zimbabwe    2005  Total (recorded+unrecorded) alcohol per capita...   
2784     Zimbabwe    2000  Total (recorded+unrecorded) alcohol per capita...   

            Dim1  First Tooltip  
0     Both sexes           0.21  
3     Both sexes           0.21  
6     Both sexes           0.21  
9     Both sexes           7.17  
12    Both sexes           6.74  
...          ...            ...  
2772  Both sexes           4.67  
2775  Both sexes           4.92  
2778  Both sexes           3.93  
2781  Both sexes           2.77  
2784  Both sexes           2.46  

[929 rows x 5 columns]

## Dataset to export

In [61]:
alcohol_both_sex_2018 = alcohol_both_sex[alcohol_both_sex['Period'] == 2018]
alcohol_both_sex_2018

Location  Period  \
0                            Afghanistan    2018   
9                                Albania    2018   
24                               Algeria    2018   
39                               Andorra    2018   
54                                Angola    2018   
...                                  ...     ...   
2712  Venezuela (Bolivarian Republic of)    2018   
2727                            Viet Nam    2018   
2742                               Yemen    2018   
2757                              Zambia    2018   
2772                            Zimbabwe    2018   

                                              Indicator        Dim1  \
0     Total (recorded+unrecorded) alcohol per capita...  Both sexes   
9     Total (recorded+unrecorded) alcohol per capita...  Both sexes   
24    Total (recorded+unrecorded) alcohol per capita...  Both sexes   
39    Total (recorded+unrecorded) alcohol per capita...  Both sexes   
54    Total (recorded+unrecorded) alcohol per capita...  Both sexes   
...                                                 ...         ...   
2712  Total (recorded+unrecorded) alcohol per capita...  Both sexes   
2727  Total (recorded+unrecorded) alcohol per capita...  Both sexes   
2742  Total (recorded+unrecorded) alcohol per capita...  Both sexes   
2757  Total (recorded+unrecorded) alcohol per capita...  Both sexes   
2772  Total (recorded+unrecorded) alcohol per capita...  Both sexes   

      First Tooltip  
0             0.210  
9             7.170  
24            0.950  
39           11.020  
54            6.940  
...             ...  
2712          4.140  
2727          8.660  
2742          0.051  
2757          6.540  
2772          4.670  

[188 rows x 5 columns]

## Dataset to export

In [62]:
alcohol_both_sex_2015 = alcohol_both_sex[alcohol_both_sex['Period'] == 2015]
alcohol_both_sex_2015

Location  Period  \
3                            Afghanistan    2015   
12                               Albania    2015   
27                               Algeria    2015   
42                               Andorra    2015   
57                                Angola    2015   
...                                  ...     ...   
2715  Venezuela (Bolivarian Republic of)    2015   
2730                            Viet Nam    2015   
2745                               Yemen    2015   
2760                              Zambia    2015   
2775                            Zimbabwe    2015   

                                              Indicator        Dim1  \
3     Total (recorded+unrecorded) alcohol per capita...  Both sexes   
12    Total (recorded+unrecorded) alcohol per capita...  Both sexes   
27    Total (recorded+unrecorded) alcohol per capita...  Both sexes   
42    Total (recorded+unrecorded) alcohol per capita...  Both sexes   
57    Total (recorded+unrecorded) alcohol per capita...  Both sexes   
...                                                 ...         ...   
2715  Total (recorded+unrecorded) alcohol per capita...  Both sexes   
2730  Total (recorded+unrecorded) alcohol per capita...  Both sexes   
2745  Total (recorded+unrecorded) alcohol per capita...  Both sexes   
2760  Total (recorded+unrecorded) alcohol per capita...  Both sexes   
2775  Total (recorded+unrecorded) alcohol per capita...  Both sexes   

      First Tooltip  
3             0.210  
12            6.740  
27            0.930  
42           11.010  
57            7.960  
...             ...  
2715          6.820  
2730          7.860  
2745          0.055  
2760          6.890  
2775          4.920  

[188 rows x 5 columns]

## Basic Sanitization

In [63]:
sani_data = pd.read_csv(WHS_path/'atLeastBasicSanitizationServices.csv')
sani_data

Location                                          Indicator  Period  \
0     Afghanistan  Population using at least basic sanitation ser...    2017   
1     Afghanistan  Population using at least basic sanitation ser...    2017   
2     Afghanistan  Population using at least basic sanitation ser...    2017   
3     Afghanistan  Population using at least basic sanitation ser...    2016   
4     Afghanistan  Population using at least basic sanitation ser...    2016   
...           ...                                                ...     ...   
9363     Zimbabwe  Population using at least basic sanitation ser...    2001   
9364     Zimbabwe  Population using at least basic sanitation ser...    2001   
9365     Zimbabwe  Population using at least basic sanitation ser...    2000   
9366     Zimbabwe  Population using at least basic sanitation ser...    2000   
9367     Zimbabwe  Population using at least basic sanitation ser...    2000   

       Dim1  First Tooltip  
0     Total          43.42  
1     Urban          62.26  
2     Rural          37.05  
3     Total          42.05  
4     Urban          59.92  
...     ...            ...  
9363  Urban          65.05  
9364  Rural          36.19  
9365  Total          45.93  
9366  Urban          65.05  
9367  Rural          36.19  

[9368 rows x 5 columns]

In [67]:
sani_data.isnull().count()

Location         9368
Indicator        9368
Period           9368
Dim1             9368
First Tooltip    9368
dtype: int64

The years 2013, 2012, and 2011 are slightly better than 2015 and 2016.

In [66]:
sani_data['Period'].unique()

array([2017, 2016, 2015, 2014, 2013, 2012, 2011, 2010, 2009, 2008, 2007,
       2006, 2005, 2004, 2003, 2002, 2001, 2000])

In [69]:
sani_data[sani_data['Period'] == 2017].count()

Location         516
Indicator        516
Period           516
Dim1             516
First Tooltip    516
dtype: int64

In [70]:
sani_data[sani_data['Period'] == 2016].count()

Location         524
Indicator        524
Period           524
Dim1             524
First Tooltip    524
dtype: int64

In [71]:
sani_data[sani_data['Period'] == 2015].count()

Location         526
Indicator        526
Period           526
Dim1             526
First Tooltip    526
dtype: int64

In [72]:
sani_data[sani_data['Period'] == 2014].count()

Location         526
Indicator        526
Period           526
Dim1             526
First Tooltip    526
dtype: int64

In [73]:
sani_data[sani_data['Period'] == 2013].count()

Location         527
Indicator        527
Period           527
Dim1             527
First Tooltip    527
dtype: int64

In [74]:
sani_data[sani_data['Period'] == 2012].count()

Location         527
Indicator        527
Period           527
Dim1             527
First Tooltip    527
dtype: int64

In [75]:
sani_data[sani_data['Period'] == 2011].count()

Location         527
Indicator        527
Period           527
Dim1             527
First Tooltip    527
dtype: int64

In [76]:
sani_data[sani_data['Period'] == 2010].count()

Location         524
Indicator        524
Period           524
Dim1             524
First Tooltip    524
dtype: int64

In [77]:
sani_data[sani_data['Period'] == 2009].count()

Location         524
Indicator        524
Period           524
Dim1             524
First Tooltip    524
dtype: int64

In [78]:
sani_data[sani_data['Period'] == 2008].count()

Location         524
Indicator        524
Period           524
Dim1             524
First Tooltip    524
dtype: int64

In [79]:
sani_data[sani_data['Period'] == 2007].count()

Location         524
Indicator        524
Period           524
Dim1             524
First Tooltip    524
dtype: int64

In [80]:
sani_data[sani_data['Period'] == 2006].count()

Location         523
Indicator        523
Period           523
Dim1             523
First Tooltip    523
dtype: int64

In [81]:
sani_data[sani_data['Period'] == 2005].count()

Location         517
Indicator        517
Period           517
Dim1             517
First Tooltip    517
dtype: int64

In [82]:
sani_data[sani_data['Period'] == 2004].count()

Location         516
Indicator        516
Period           516
Dim1             516
First Tooltip    516
dtype: int64

In [83]:
sani_data[sani_data['Period'] == 2003].count()

Location         516
Indicator        516
Period           516
Dim1             516
First Tooltip    516
dtype: int64

In [84]:
sani_data[sani_data['Period'] == 2002].count()

Location         516
Indicator        516
Period           516
Dim1             516
First Tooltip    516
dtype: int64

In [85]:
sani_data[sani_data['Period'] == 2001].count()

Location         510
Indicator        510
Period           510
Dim1             510
First Tooltip    510
dtype: int64

In [86]:
sani_data[sani_data['Period'] == 2000].count()

Location         501
Indicator        501
Period           501
Dim1             501
First Tooltip    501
dtype: int64

In [89]:
sani_data['Indicator'].unique()

array(['Population using at least basic sanitation services (%)'],
      dtype=object)

In [90]:
sani_data_tot = sani_data[sani_data['Dim1'] == "Total"]
sani_data_tot

Location                                          Indicator  Period  \
0     Afghanistan  Population using at least basic sanitation ser...    2017   
3     Afghanistan  Population using at least basic sanitation ser...    2016   
6     Afghanistan  Population using at least basic sanitation ser...    2015   
9     Afghanistan  Population using at least basic sanitation ser...    2014   
12    Afghanistan  Population using at least basic sanitation ser...    2013   
...           ...                                                ...     ...   
9353     Zimbabwe  Population using at least basic sanitation ser...    2004   
9356     Zimbabwe  Population using at least basic sanitation ser...    2003   
9359     Zimbabwe  Population using at least basic sanitation ser...    2002   
9362     Zimbabwe  Population using at least basic sanitation ser...    2001   
9365     Zimbabwe  Population using at least basic sanitation ser...    2000   

       Dim1  First Tooltip  
0     Total          43.42  
3     Total          42.05  
6     Total          40.71  
9     Total          39.37  
12    Total          38.06  
...     ...            ...  
9353  Total          45.37  
9356  Total          46.11  
9359  Total          46.15  
9362  Total          46.05  
9365  Total          45.93  

[3439 rows x 5 columns]

## Dataset to export

In [93]:
sani_data_2013 = sani_data_tot[sani_data_tot['Period'] == 2013]
sani_data_2013

Location  \
12                           Afghanistan   
66                               Albania   
120                              Algeria   
174                              Andorra   
228                               Angola   
...                                  ...   
9143  Venezuela (Bolivarian Republic of)   
9164                            Viet Nam   
9218                               Yemen   
9272                              Zambia   
9326                            Zimbabwe   

                                              Indicator  Period   Dim1  \
12    Population using at least basic sanitation ser...    2013  Total   
66    Population using at least basic sanitation ser...    2013  Total   
120   Population using at least basic sanitation ser...    2013  Total   
174   Population using at least basic sanitation ser...    2013  Total   
228   Population using at least basic sanitation ser...    2013  Total   
...                                                 ...     ...    ...   
9143  Population using at least basic sanitation ser...    2013  Total   
9164  Population using at least basic sanitation ser...    2013  Total   
9218  Population using at least basic sanitation ser...    2013  Total   
9272  Population using at least basic sanitation ser...    2013  Total   
9326  Population using at least basic sanitation ser...    2013  Total   

      First Tooltip  
12            38.06  
66            97.31  
120           87.16  
174          100.00  
228           44.85  
...             ...  
9143          93.94  
9164          76.49  
9218          56.31  
9272          25.74  
9326          38.94  

[194 rows x 5 columns]

## Dataset to export

In [94]:
sani_data_2012 = sani_data_tot[sani_data_tot['Period'] == 2012]
sani_data_2012

Location  \
15                           Afghanistan   
69                               Albania   
123                              Algeria   
177                              Andorra   
231                               Angola   
...                                  ...   
9144  Venezuela (Bolivarian Republic of)   
9167                            Viet Nam   
9221                               Yemen   
9275                              Zambia   
9329                            Zimbabwe   

                                              Indicator  Period   Dim1  \
15    Population using at least basic sanitation ser...    2012  Total   
69    Population using at least basic sanitation ser...    2012  Total   
123   Population using at least basic sanitation ser...    2012  Total   
177   Population using at least basic sanitation ser...    2012  Total   
231   Population using at least basic sanitation ser...    2012  Total   
...                                                 ...     ...    ...   
9144  Population using at least basic sanitation ser...    2012  Total   
9167  Population using at least basic sanitation ser...    2012  Total   
9221  Population using at least basic sanitation ser...    2012  Total   
9275  Population using at least basic sanitation ser...    2012  Total   
9329  Population using at least basic sanitation ser...    2012  Total   

      First Tooltip  
15            36.75  
69            96.78  
123           86.97  
177          100.00  
231           43.59  
...             ...  
9144          93.94  
9167          74.68  
9221          55.19  
9275          25.61  
9329          39.64  

[194 rows x 5 columns]